src/frontend/Symbol_table.mli

```diff
@@ -2,32 +2,32 @@
 
 type 'a state
```

Comment:
I think this might be supposed to be called `t` if we follow the Jane Street style, but separately if we are doing it functionally do we even need a Symbol table module at all or should we just be using a map or set?

Reply:
My sense is that we could probably go both ways.

The current approach where we use a combination of a map and a stack (and some sets) clearly works, in which case it makes sense to wrap things up in a module.
Alternatively, when working purely functionally people often use the call-stack to do the work of the stack data structure in the symbol table. However, then, it might still be useful to keep the sets around in this case.

Overall, my sense is that it might be useful to keep the Symbol table its own abstract type (module) as that gives you more flexibility regarding implementation and it puts more constraints in place to make sure people don't use arbitrary destructive updates on it.